soap/soap.php

```diff
@@ -261,7 +262,135 @@ public function SoapClient ($wsdl, array $options = null) {}
      * @throws SoapFault A SoapFault exception will be thrown if the wsdl URI cannot be loaded.
      * @since 5.0.1
      */
-    public function __construct ($wsdl, array $options = null) {}
+	#[PhpStormStubsElementAvailable(to: '5.6')]
```

Comment:
Could you please clarify why you choose `PhpStormStubsElementAvailable` but not `LanguageLevelTypeAware` here? Type hints in parameters could be customized using the latter one.

Reply:
can LanguageLevelTypeAware use multiple different versions? as 7.0 and 7.1 have differencies also. My commit did not yet take those into account.

Reply:
meaning whole file changes.

Reply:
Yes, sure. Multiple versions can be declared like `#[LanguageLevelTypeAware(['8.0' => 'bool', "7.0" => 'int', '7.1' => 'string'], default: 'float')]`

Reply:
how about default changing? it has changed from null to '' in some cases. Sorry if asking basics, but did not found any documentation for that.

Reply:
The default value doesn't matter too much actually as this is just a stub and if a parameter will not be provided in the code, the default value from the extension will be used in runtime. So if the extension has default value `''` for some parameter then in runtime `''` will be used despite stub will be declared with default value `null`. But it worth mentioning that we still have a separate test that compares default values of parameters with reflection for sake of stubs cleanliness

Reply:
I'd object that PHP EA has an inspection that suggest dropping a literal parameter that has the same value as the default.

This means it could make a wrong suggestion if the default value is not the right one

Reply:
>This means it could make a wrong suggestion if the default value is not the right one
Yes, that's right. From the inspection's point of view, a suggestion can be wrong depending on the language version. But unfortunately, we don't support LanguageLevelAwareness for default values yet to properly process such cases.

Reply:
Would using  PhpStormStubsElementAvailable solve that?

Reply:
Does LanguageLevelTypeAware support return types?

Reply:
Can't say about EA but regarding PhpStorm the answer is "No". At least not yet. If you think it would be useful could you please create a ticket in YouTrack to highlight the redundant optional argument depending on `PhpStormStubsElementAvailable` attribute?

Reply:
changed the PR to use LanguageLevelTypeAware